skoolkit/snapinfo.py

```diff
@@ -426,8 +429,11 @@ def main(args):
         _find_text(infile, namespace.text)
     elif namespace.peek is not None:
         _peek(infile, namespace.peek)
-    elif namespace.basic:
-        print(BasicLister().list_basic(get_snapshot(infile)))
+    elif namespace.basic or namespace.variables:
+        if namespace.basic:
+            print(BasicLister().list_basic(get_snapshot(infile)))
+        if namespace.variables:
+            print(VariableLister().list_variables(get_snapshot(infile)))
```

Comment:
It's possible that get_snapshot() will be called twice here - once more than is necessary.